src/Compilers/CSharp/Test/Semantic/Semantics/Utf8StringsLiteralsTests.cs

```diff
@@ -0,0 +1,1790 @@
+
+        System.Console.WriteLine(x);
+        System.Console.WriteLine(y);
+        System.Console.WriteLine(z);
+    }
+}
+
+class C1
+{
+    public static implicit operator C1(byte[] x)
+    {
+        return new C1();
+    }
+}
+
+class C2
+{
+    public static implicit operator C2(Span<byte> x)
+    {
+        return new C2();
+    }
+}
+
+class C3
+{
+    public static implicit operator C3(ReadOnlySpan<byte> x)
+    {
+        return new C3();
+    }
+}
+";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.DebugExe);
+
+            // PROTOTYPE(UTF8StringLiterals) : Confirm this is the behavior we want. Or should we disallow this conversion in expression tree?
+            CompileAndVerify(comp, expectedOutput: @"
+() => Convert(new [] {104, 101, 108, 108, 111}, C1)
+() => Convert(new Span`1(new [] {100, 111, 103}), C2)
+() => Convert(new ReadOnlySpan`1(new [] {99, 97, 116}), C3)
+").VerifyDiagnostics();
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void OverloadResolution_01()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        System.Console.WriteLine(Test(""s""));
+    }
+
+    static string Test(ReadOnlySpan<char> a) => ""ReadOnlySpan"";
+    static string Test(byte[] a) => ""array"";
+}
+";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.DebugExe);
+
+            CompileAndVerify(comp, expectedOutput: @"ReadOnlySpan").VerifyDiagnostics();
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void OverloadResolution_02()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        System.Console.WriteLine(Test(""s""));
+    }
+
+    static string Test(byte[] a) => ""array"";
+    static string Test(ReadOnlySpan<char> a) => ""ReadOnlySpan"";
+}
+";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.DebugExe);
+
+            CompileAndVerify(comp, expectedOutput: @"ReadOnlySpan").VerifyDiagnostics();
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void OverloadResolution_03()
+        {
+            var source = @"
+class C
+{
+    static void Main()
+    {
+        System.Console.WriteLine(Test(""s""));
+    }
+
+    static string Test(C1 a) => ""string"";
+    static string Test(C2 a) => ""array"";
+}
+
+class C1
+{
+    public static implicit operator C1(string x)
+    {
+        return new C1();
+    }
+}
+
+class C2
+{
+    public static implicit operator C2(byte[] x)
+    {
+        return new C2();
+    }
+}
+";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.DebugExe);
+
+            CompileAndVerify(comp, expectedOutput: @"string").VerifyDiagnostics();
+        }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

Consider including an example where overload resolution behavior has changed. For instance, will the following report "string" with C#10, and "byte[]" with C#11?

```csharp
using System;

class Program
{
    static void Main()
    {
        var p = new Program();
        Console.WriteLine(p.M(""));
    }

    public string M(byte[] b) => "byte[]";
}

static class E
{
    public static string M(this object o, string s) => "string";
}
```